platform-sdk/docs/hashgraph_glossary.md

```diff
@@ -0,0 +1,250 @@
+  transactions in a round are handled, a new roster is added for the future. An event is only allowed to be in the
+  hashgraph when its birth round is the roster round for some roster in the queue. And it must have been created by a
+  node that is listed in that roster. Of course, ancient, non-expired events may still be in memory, to gossip to nodes
+  that are behind this node, but they don't count as being in this node's hashgraph. Events with birth rounds too far in
+  the future should not be accepted during gossip. And a node should not send such events to another node, if it knows
+  that node will just discard them for that reason.
+
+  [top](#top)
+
+<a name="fields-of-an-event"/>
+
+## Fields of an event
+
+- __birthRound__ - set to the creator's pending round at the moment of creation. Is signed and immutable.
+- __votingRound__ - the round for this event, which determines whether it is a witness. It is a witness if its voting
+  round is different from its self-parent's voting round (or it has no self-parent). If it is a witness, then it might
+  be eligible for election as a judge (if it's in the pending round), or it might be an initial voter for judge (if it's
+  in the pending round plus `d`), or it might be a voter and vote collector (if it's later than the pending round plus
+  `d`). Every time the processing of a new round starts, the voting round is recalculated for all events. It can be
+  different than it was in the past because non-ancestors of the latest judges are defined to have a voting round of
+  negative infinity (represented as a 0 in memory), and that can cause all their descendents to change their voting
+  rounds. If a judge in the consensus round is not a descendant of any other judge in the consensus round, then it is
+  guaranteed to have a voting round equal to the consensus round. In the fields listed here, this is the only one that
+  is mutable. All the rest are immutable, and are signed when the event is created.
+- __consensusRound__ - the first round in which this event was an ancestor of all judges.
+- __consensusTimestamp__ - the median timestamp of when it first reached each of the nodes that created judges in its
+  consensus round. This is adjusted by adding nanoseconds to ensure that in consensus order, each transaction is at
+  least 1000 nanoseconds after the previous one. (That ensures they are unique and monotonically increasing, and there
+  are big enough gaps so that synthetic transactions inserted between them can have timestamps that are unique).
+- __consensusOrder__ - an long `N` indicating this is the `N`th event in all of history, according to the calculated
+  consensus order.
+- __selfParent__ - the hash of the self parent (or null if there is none, or the self-parent's birth round is ancient)
+- __selfParentCreator__ - the node that created the self-parent event. If this is incorrect, then it cannot be added to
+  the hashgraph until the self-parent becomes ancient.
```

Comment:
It's a bit unclear what `If this is incorrect` means exactly. I suggest to reword this somehow differently.